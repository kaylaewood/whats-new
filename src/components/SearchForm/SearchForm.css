header {
  align-items: center;
  background-color: #333332;
  box-shadow: 0 0 15px 0 #F1F1F1;
  display: flex;
  grid-area: searchBar;
  height: 5em;
  justify-content: space-between;
  padding: 2em;
  position: fixed;
  width: 100%;
  z-index: 100;
}

h1 {
  font-family: 'Abril Fatface', cursive;
  font-size: 2.5em;
}

span {
  color: #EC5449;
  font-family: 'Abril Fatface', cursive;
}

.search-section {
  width: 40%;
}

input {
  font-size: 1em;
  padding: .7em;
  padding-left: 6px;
  width: 65%;
  border: none;
}

.search-button {
  background-color: #EC5449;
  border: none;
  color: white;
  font-size: 1em;
  font-weight: bold;
  margin-left: 1%;
  padding: .7em;
  width: 34%;
}

@media only screen and (max-width: 900px) {
  h1 {
    font-size: 1.5em;
  }

  .search-section {
    width: 55%;
  }
  input,
  .search-button {
    font-size: .8em;
  }

}

@media only screen and (max-width: 745px) {
  header {
    flex-direction: column;
    height: 7em;
    padding: 1em;
  }
  .search-section {
    width: 80%;
  }
}
