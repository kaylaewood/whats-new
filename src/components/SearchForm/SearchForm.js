import React from 'react';
import './SearchForm.css';

function SearchForm() {
  return (
    <div>
      <p>This is the search bar.</p>
    </div>
  )
}

export default SearchForm;
