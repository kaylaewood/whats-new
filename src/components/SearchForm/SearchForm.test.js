import React from 'react';
import ReactDOM from 'react-dom';
import SearchForm from './SearchForm';
import { shallow } from 'enzyme';

describe('SearchForm', () => {
  it('should correctly match the snapshot with all the data passed correctly', () => {
    const wrapper = shallow(<SearchForm />);
    expect(wrapper).toMatchSnapshot();
  })

  it('should call searchNews when button is clicked', () => {
    const mockSearchNews = jest.fn();
    const wrapper = shallow(<SearchForm searchNews = {mockSearchNews} />);
    const document = {
      id: 'search-input',
      value: 'search query'
    };

    wrapper.find('button').simulate('click');
    expect(mockSearchNews).toHaveBeenCalled();
  })
});
