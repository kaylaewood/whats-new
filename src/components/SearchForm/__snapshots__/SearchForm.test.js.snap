// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm should correctly match the snapshot with all the data passed correctly 1`] = `
<header>
  <h1>
    what's 
    <span
      className="heading-highlight"
    >
      new
    </span>
     in your world?
  </h1>
  <form
    className="search-section"
  >
    <input
      id="search-input"
      onChange={[Function]}
      placeholder="Search for news articles here"
      type="textBox"
    />
    <button
      className="search-button"
      onClick={[Function]}
    >
      SEARCH
    </button>
  </form>
</header>
`;
